packages/plugins/src/hardhat/deployments.ts

```diff
@@ -152,16 +154,37 @@ export const getContract = async (
   }
 
   const userConfig = userConfigs[0]
+  const contractConfig = userConfig.contracts[referenceName]
 
-  const proxyAddress =
-    userConfig.contracts[referenceName].externalProxy ||
+  let address =
+    contractConfig.externalProxy ||
     getDefaultProxyAddress(userConfig.options.projectName, referenceName)
-  if ((await isContractDeployed(proxyAddress, hre.ethers.provider)) === false) {
+
+  if (contractConfig.kind === 'no-proxy') {
+    const { abi, bytecode } = hre.artifacts.readArtifactSync(
+      contractConfig.contract
+    )
+    const creationCodeWithConstructorArgs = getCreationCodeWithConstructorArgs(
+      bytecode,
+      contractConfig.constructorArgs || {},
+      referenceName,
+      abi
+    )
+    address = getContractAddress(
+      userConfig.options.projectName,
+      creationCodeWithConstructorArgs
+    )
+  }
```

Comment:
IMO it's slightly cleaner to only assign the value of `address` once instead of re-assigning it if we're dealing with an unproxied contract